### PR TITLE
Add Gen.single and Gen.decimal

### LINKED ISSUE #249: Generate single/float32
I noticed the auto-generator in Hegehog.Experimental failing for one of my record types. It was because it couldn't generate a `single` (`float32`) property. When I tried to add support for it to the auto-generator, I discovered that Hedgehog has no `Gen.single`. Can this be added?

In the meantine, is there any downside to doing this in the auto-generator? (Or if you can implement `Gen.single` quickly, I can wait.)

```f#
Gen.float (Range.exponentialFrom 0. (float Single.MinValue) (float Single.MaxValue)) |> Gen.map single
```

### PR DESCRIPTION
**Pull request by cmeeren**
Fixes #249

### CODE REVIEW

File: src/Hedgehog/Gen.fs [hunk 1]
```diff
@@ -3,12 +3,17 @@
...
+
 /// A generator for values and shrink trees of type 'a.
 [<Struct>]
 type Gen<'a> =
     | Gen of Random<Tree<'a>>
 
 module Gen =
+
+    // Needed because 'double' is shadowed later
+    let inline private fsharpDouble x = double x
```
Comment (moodmosaic) [inline src/Hedgehog/Gen.fs:15]: Can't we just use the full qualified name instead? If we have to do that in more than three places, perhaps we can declare an inline function as the one above. Right now it's been used only in two places.

File: src/Hedgehog/Gen.fs [hunk 2]
```diff
@@ -3,12 +3,17 @@
 open System
 open Hedgehog.Numeric
 
+
```
Comment (moodmosaic) [inline src/Hedgehog/Gen.fs:6]: What's the purpose of that extra line?

**Review by moodmosaic — COMMENTED** [review]
Looking good. Left a few comments.

File: src/Hedgehog/Gen.fs (hunk 1 above)
Comment (cmeeren) [reply src/Hedgehog/Gen.fs:15]: Sure. I didn't find where it was (I looked in `FSharp.Core.Operators`, but it wasn't there), and decided that it wasn't worth the trouble. Do you know the fully qualified name?

File: src/Hedgehog/Gen.fs (hunk 2 above)
Comment (cmeeren) [reply src/Hedgehog/Gen.fs:6]: Absolutely none. I'll remove that rogue edit. It was caused by my adding `fsharpDouble` there first before realizing it had to be moved, and then I forgot to check the diff before I committed.

File: src/Hedgehog/Gen.fs (hunk 1 above)
Comment (cmeeren) [reply src/Hedgehog/Gen.fs:15]: Nevermind, I found it.

**Comment by cmeeren** [comment]
Now the diff is cleaner.

**Review by moodmosaic — APPROVED** [review]
:+1:

**Pull request merged**